src/V3Ast.cpp

```diff
@@ -1111,68 +1111,6 @@ bool AstNode::sameTreeIter(const AstNode* node1p, const AstNode* node2p, bool ig
             && (ignNext || sameTreeIter(node1p->m_nextp, node2p->m_nextp, false, gateOnly)));
 }
 
-//======================================================================
-// Debugging
-
-void AstNode::checkTreeIter(const AstNode* prevBackp) const VL_MT_STABLE {
```

Comment:
AstNode:: base functions are typically in V3Ast.cpp, as are the other Iter() functions, so perhaps not move this?

Reply:
The check function needs downcast or VN_IS/VN_CAST.
AstNodeExpr and its derived classes are not visible in V3Ast.cpp

If it is ok to include V3AstNodeExpr.h or generated header file in V3Ast.cpp, I will do so.

Reply:
V3Ast.cpp already includes V3PchAstMt.h which includes V3Ast.h which includes V3AstNodeExpr.h, so I don't understand.

Reply:
comment glitch.

You are right. When I initially wrote in V3Ast.cpp, I saw a compiler error, but I must have typo.
It compiles.

I will move both functions to V3Ast.cpp